Index: Lib/Protocols/IdSSLOpenSSLHeaders.pas
===================================================================
--- Lib/Protocols/IdSSLOpenSSLHeaders.pas	(revision 5369)
+++ Lib/Protocols/IdSSLOpenSSLHeaders.pas	(working copy)
@@ -16614,12 +16614,18 @@
   X509_get_default_cert_file : function : PIdAnsiChar cdecl = nil;
   {$EXTERNALSYM X509_get_default_cert_file_env}
   X509_get_default_cert_file_env : function : PIdAnsiChar cdecl = nil;
+  {$EXTERNALSYM BIO_set_flags}
+  BIO_set_flags : procedure(b: PBIO; flags: integer) cdecl = nil;
   {$EXTERNALSYM BIO_new}
   BIO_new : function(_type: PBIO_METHOD): PBIO cdecl = nil;
   {$EXTERNALSYM BIO_new_mem_buf}
   BIO_new_mem_buf : function (buf : Pointer; len : TIdC_INT) : PBIO cdecl = nil;
+  {$EXTERNALSYM BIO_push}
+  BIO_push : function(b: PBIO; append: PBIO): PBIO cdecl = nil;
   {$EXTERNALSYM BIO_free}
   BIO_free : function(bio: PBIO): TIdC_INT cdecl = nil;
+  {$EXTERNALSYM BIO_free_all}
+  BIO_free_all : procedure(bio: PBIO) cdecl = nil;
   {$EXTERNALSYM BIO_s_mem}
   BIO_s_mem : function: PBIO_METHOD cdecl = nil;
   {$EXTERNALSYM BIO_s_file}
@@ -19670,6 +19676,7 @@
   {CH fn_BUF_MEM_grow = 'BUF_MEM_grow'; }  {Do not localize}
   {CH fn_BUF_strdup = 'BUF_strdup'; }  {Do not localize}
   {CH fn_ERR_load_BUF_strings = 'ERR_load_BUF_strings'; }  {Do not localize}
+  fn_BIO_set_flags = 'BIO_set_flags';  {Do not localize}
   {CH fn_BIO_ctrl_pending = 'BIO_ctrl_pending'; }  {Do not localize}
   {CH fn_BIO_ctrl_wpending = 'BIO_ctrl_wpending'; }  {Do not localize}
   {CH fn_BIO_ctrl_get_write_guarantee = 'BIO_ctrl_get_write_guarantee'; }  {Do not localize}
@@ -19691,9 +19698,9 @@
   fn_BIO_ctrl = 'BIO_ctrl';  {Do not localize}
   fn_BIO_ptr_ctrl = 'BIO_ptr_ctrl';   {Do not localize}
   fn_BIO_int_ctrl = 'BIO_int_ctrl';  {Do not localize}
-  {CH fn_BIO_push = 'BIO_push'; }  {Do not localize}
+  fn_BIO_push = 'BIO_push';   {Do not localize}
   {CH fn_BIO_pop = 'BIO_pop'; }  {Do not localize}
-  {CH fn_BIO_free_all = 'BIO_free_all'; }  {Do not localize}
+  fn_BIO_free_all = 'BIO_free_all';   {Do not localize}
   {CH fn_BIO_find_type = 'BIO_find_type'; }  {Do not localize}
   {CH fn_BIO_get_retry_BIO = 'BIO_get_retry_BIO'; }  {Do not localize}
   {CH fn_BIO_get_retry_reason = 'BIO_get_retry_reason'; }  {Do not localize}
@@ -22867,8 +22874,11 @@
   @BN_bn2hex := LoadFunctionCLib(fn_BN_bn2hex,False);
   @BN_set_word := LoadFunctionCLib(fn_BN_set_word,False);
   //BIO
+  @BIO_set_flags := LoadFunctionCLib(fn_BIO_set_flags);
   @BIO_new := LoadFunctionCLib(fn_BIO_new);   //Used by Indy
+  @BIO_push := LoadFunctionCLib(fn_BIO_push);
   @BIO_free := LoadFunctionCLib(fn_BIO_free);  //Used by Indy
+  @BIO_free_all := LoadFunctionCLib(fn_BIO_free_all);
   @BIO_new_mem_buf := LoadFunctionCLib(fn_BIO_new_mem_buf);   //Used by Indy
   @BIO_s_mem := LoadFunctionCLib(fn_BIO_s_mem);  //Used by Indy
   @BIO_s_file := LoadFunctionCLib(fn_BIO_s_file,False);
@@ -23605,8 +23615,11 @@
   //DH
   @DH_free := nil;
   //BIO
+  @BIO_set_flags := nil;
   @BIO_new := nil;
+  @BIO_push := nil;
   @BIO_free := nil;
+  @BIO_free_all := nil;
   @BIO_s_mem := nil;
   @BIO_s_file := nil;
   @BIO_set_ex_data := nil;
